### DISCUSSION TITLE
Is there a way we can do ```passedValidation``` or ```after()``` feature ?

### DISCUSSION THREAD

**Discussion by hvsharma63** [spatie/laravel-data#784] [Q&A]
I'm looking for a way to implement [Performing Additional Validation](https://laravel.com/docs/11.x/validation#performing-additional-validation-on-form-requests) and [Preparing Input for Validation](https://laravel.com/docs/11.x/validation#preparing-input-for-validation) using laravel-data. 

I checked the documentation but couldn't find anything. 

Is there a way? 
Thanks.

**Comment by 200-0K** [comment]
You can use `withValidator` to do extra validation:

```php
use Illuminate\Validation\Validator;
use Spatie\LaravelData\Data;

class SongData extends Data
{
    // ...

    public static function withValidator(Validator $validator): void
    {
        // This will run after the validator has checked the rules.
        $validator->after(function (Validator $validator) {
            if ($validator->errors()->isEmpty()) {
                // Extra Validation
            }
        });
    }
}
```
